examples/rotation_bin/rotation_bin/mod.rs

```diff
@@ -0,0 +1,100 @@
+mod imp;
+
+use gtk::glib;
+use gtk::subclass::prelude::*;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, glib::Enum)]
+#[repr(i32)]
+#[enum_type(name = "RotationEnum")]
+pub enum Rotation {
+    Normal = 0,
+    Deg90 = 90,
+    Deg180 = 180,
+    Deg270 = 270,
+}
+
+impl Rotation {
+    pub fn rotate_clockwise(self) -> Self {
+        match self {
+            Self::Normal => Self::Deg90,
+            Self::Deg90 => Self::Deg180,
+            Self::Deg180 => Self::Deg270,
+            Self::Deg270 => Self::Normal,
+        }
+    }
+
+    pub fn rotate_counter_clockwise(self) -> Self {
+        match self {
+            Self::Normal => Self::Deg270,
+            Self::Deg90 => Self::Normal,
+            Self::Deg180 => Self::Deg90,
+            Self::Deg270 => Self::Deg180,
+        }
+    }
+}
+
+impl From<i32> for Rotation {
+    fn from(item: i32) -> Self {
```

Comment:
Do you actually need this somewhere? If you do, as it's part of the public API of the widget this should really be a `TryFrom`.

Reply:
So this isn't needed somewhere, but its useful if there are apis that give you ints and you want to parse them to something usable. Indeed a TryFrom would have been more appropriate with arbitrary input. I removed it for now since its not used.